drivers/spi/spi_nrfx_spim.c

```diff
@@ -378,18 +387,24 @@ static int transceive(const struct device *dev,
 
 	spi_context_lock(&dev_data->ctx, asynchronous, cb, userdata, spi_cfg);
 
+	(void)pm_device_runtime_get(dev);
+
 	error = configure(dev, spi_cfg);
-	if (error == 0) {
-		dev_data->busy = true;
+	if (error < 0) {
+		(void)pm_device_runtime_put(dev);
+		goto release;
```

Comment:
Single release point, this is a fairly common pattern when you always need to release something if there's an error. Here we'd need 2 calls to spi_context_release otherwise

Reply:
In Zephyr we tend to branch for the error condition, not success. This avoids the need for indenting when things succeed, so code tends to be more readable: success path (normally the interesting path) is always at the same indentation level. In such case, the goto avoids having 2 calls.

Reply:
Such modification is not needed in this PR and in fact it obscures the actual changes required to support the runtime PM. IMO it's better to revert it.